src/jvmMain/kotlin/controller/validation/GuavaSessionCacheAdapter.kt

```diff
@@ -8,7 +8,11 @@ import java.util.*
 import java.util.concurrent.TimeUnit
 
 class GuavaSessionCacheAdapter(cacheSize : Long, cacheDuration: Long) : SessionCache {
-    private val cache: Cache<String, ValidationEngine> = CacheBuilder.newBuilder().expireAfterAccess(cacheDuration, TimeUnit.MINUTES).maximumSize(cacheSize).build()
+    private val cache: Cache<String, ValidationEngine> = if (cacheDuration > 0) {
+        CacheBuilder.newBuilder().expireAfterAccess(cacheDuration, TimeUnit.MINUTES).maximumSize(cacheSize).build()
+    } else {
+        CacheBuilder.newBuilder().maximumSize(cacheSize).build()
+    }
```

Comment:
Seeing this now, I feel like this new "if" all by itself is enough to support the "never expire" concept and we don't even need to offer both cache implementation options. But I'll leave that up to you.

Could we also make the maximumSize optional, something like this? The default if the env var isn't set is 4 so this should only allow for more options, not change any existing behavior.
```suggestion
    private val cache: Cache<String, ValidationEngine>;
    init {
        val cacheBuilder = CacheBuilder.newBuilder()
        if (cacheDuration > 0) {
            cacheBuilder.expireAfterAccess(cacheDuration, TimeUnit.MINUTES)
        }
        if (cacheSize > 0) {
            cacheBuilder.maximumSize(cacheSize)
        }
        cache = cacheBuilder.build<String, ValidationEngine>()
    }
```

Reply:
I see what you mean with regards to not offering both implementation options. I'll be happy to drop support for the original implementation, but I'll first mark it as deprecated for a while before dropping it.